awscli/argprocess.py

```diff
@@ -188,13 +189,20 @@ def _unpack_cli_arg(argument_model, value, cli_name):
         return six.text_type(value)
 
 
-def _unpack_json_cli_arg(argument_model, value, cli_name):
-    try:
-        return json.loads(value, object_pairs_hook=OrderedDict)
-    except ValueError as e:
-        raise ParamError(
-            cli_name, "Invalid JSON: %s\nJSON received: %s"
-            % (e, value))
+def _unpack_json_cli_arg(value, cli_name):
+    if value.lstrip().startswith("---"):
+        try:
+            return yaml.load(value)
+        except Exception as e:
+            raise ParamError(cli_name, "Invalid JSON or YAML: %s\nReceived: %s"
```

Comment:
If the file starts with `---`, can the data ever be JSON? If not, I'd like to suggest that the error message is changed.